backend/src/auth/tokens.go

```diff
@@ -0,0 +1,191 @@
+package auth
+
+import (
+	"time"
+
+	"github.com/GenerateNU/sac/backend/src/config"
+	"github.com/GenerateNU/sac/backend/src/errors"
+	"github.com/GenerateNU/sac/backend/src/types"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/golang-jwt/jwt"
+)
+
+func CreateTokenPair(id string, role string) (*string, *string, *errors.Error) {
+	accessToken, catErr := CreateAccessToken(id, role)
+	if catErr != nil {
+		return nil, nil, catErr
+	}
+
+	refreshToken, crtErr := CreateRefreshToken(id)
+	if crtErr != nil {
+		return nil, nil, crtErr
+	}
+
+	return accessToken, refreshToken, nil
+}
+
+// CreateAccessToken creates a new access token for the user
+func CreateAccessToken(id string, role string) (*string, *errors.Error) {
+	if id == "" || role == "" {
+		return nil, &errors.FailedToCreateAccessToken
+	}
+
+	var settings config.Settings
+
+	accessTokenClaims := jwt.NewWithClaims(jwt.SigningMethodHS256, &types.CustomClaims{
+		StandardClaims: jwt.StandardClaims{
+			IssuedAt:  time.Now().Unix(),
+			Issuer:    id,
+			ExpiresAt: time.Now().Add(time.Minute * 15).Unix(),
+		},
+		Role: role,
+	})
+
+	accessToken, err := SignToken(accessTokenClaims, settings.AuthSecret.AccessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	return accessToken, nil
+}
+
+// CreateRefreshToken creates a new refresh token for the user
+func CreateRefreshToken(id string) (*string, *errors.Error) {
+	if id == "" {
+		return nil, &errors.FailedToCreateRefreshToken
+	}
+
+	var settings config.Settings
+
+	refreshTokenClaims := jwt.NewWithClaims(jwt.SigningMethodHS256, &jwt.StandardClaims{
+		IssuedAt:  time.Now().Unix(),
+		Issuer:    id,
+		ExpiresAt: time.Now().Add(time.Hour * 24 * 30).Unix(),
+	})
+
+	refreshToken, err := SignToken(refreshTokenClaims, settings.AuthSecret.RefreshToken)
+	if err != nil {
+		return nil, err
+	}
+
+	return refreshToken, nil
+}
+
+func SignToken(token *jwt.Token, secret string) (*string, *errors.Error) {
+	tokenString, err := token.SignedString([]byte(secret))
+	if err != nil {
+		return nil, &errors.FailedToSignToken
+	}
+	return &tokenString, nil
+}
+
+// CreateCookie creates a new cookie
+func CreateCookie(name string, value string, expires time.Time) *fiber.Cookie {
+	return &fiber.Cookie{
+		Name:     name,
+		Value:    value,
+		Expires:  expires,
+		HTTPOnly: true,
+	}
+}
+
+// ExpireCookie expires a cookie
+func ExpireCookie(name string) *fiber.Cookie {
+	return &fiber.Cookie{
+		Name:     name,
+		Value:    "",
+		Expires:  time.Now().Add(-time.Hour),
```

Comment:
Same as above

Reply:
was talking about making it configurable, that actually doesn't apply here after second thought